HHS/simpler-grants-gov
# [Issue 2448] use v1 search endpoint and frontend fetch pattern refactors

## Summary
Fixes #2448

### Time to review: __45 mins__

## Changes proposed
**DISCLAIMER**: there is a lot of scope creep in this PR. Feel free to push back on anything you feel goes to far here.

The primary purpose of this change is to change the search API to use the `v1` version of the search endpoint in order to activate the use of Open Search on the backend. This purpose is accomplished in the first commit, which can be split into a separate PR or commit if called for.

The rest of the PR contains refactors to the `API` fetch system that:

* should reduce redundancy and make better use of the classical setup in place here
* makes better use of available TS functionality
* better encapsulates domain specific logic (eg moving search specific functionality out of the base class)
* fixes bugs

## Context for reviewers
### Test Steps

1. visit the search page, and perform all manner of searches with different filters, queries, and sort options
2. _VALIDATE_: everything works as expected
3. visit various opportunity pages
4. _VALIDATE_: everything works as expected

## Additional information
> Screenshots, GIF demos, code examples or output to help show the changes working as expected.
